File: common/game.py
from enum import Enum


class Type(str, Enum):
    TIC_TAC_TOE = 'tic_tac_toe'


class Result(str, Enum):
    TBD = 'tbd'
    DRAW = 'draw'
    P1_WON = 'p1_won'
    P2_WON = 'p2_won'


class Status(str, Enum):
    IN_QUEUE = 'in_queue'
    IN_PROGRESS = 'in_progress'
    ABANDONED = 'abandoned'
    FINISHED = 'finished'

File: services/ui/src/types/game.ts
type GameType = "tic_tac_toe"

type GameResult = "tbd" | "draw" | "p1_won" | "p2_won"

type GameStatus = "in_queue" | "in_progress" | "abandoned" | "finished"

type Game = {
    id: number
    type: GameType
    result: GameResult
    status: GameStatus
    player1: number
    player2: number | null
    current_player: number
    turn_number: number
    created_at: string
    updated_at: string
}

interface GameCreate {
    type: GameType
}

export type { Game, GameType, GameResult, GameStatus, GameCreate }
